### ISSUE TITLE
workflow for fixing osmose issues

### ISSUE THREAD

**Issue by khlsvr** [MarcusWolschon/osmeditor4android#1422]
This is not probably a feature request but a question how to do the following properly. 

tldr; Saving a business with just a name tag creates two separate osmose bug tags. Then fixing those bug tags it seems like I would need to first close and "save" them both and then open them again and "upload" them both, meaning I need to open the tags 4 times.

So I am on the road and see businesses not yet added to osm. I have time to quickly add them to the map with the name tag and perhaps a contact detail like phone number. I also do other editing on the same run, but these being "complete" edits. I then upload a changeset. Later with better time I search and find out what the business is and add the proper tags. I might edit the proper tags many days later, and in the meanwhile do some other editing.

When I get back to the businesses with just a name tag, I will be seeing two separate osmose bug icons on the map at the same location with the business node. One says "Osmose: Untagged named object" and the other  "Osmose: missing tag incomplete object". Now I don't know what I should be doing here after I have properly tagged the business. Currently depending how much I want to spend time fixing these osmose bug tags I either don't do anything to them, or I change one of them from "open" to "closed" and press "save", or I fix one of them and also after pressing save I open it again and this time press "upload". And then if I want to spend even more time I do either of those two things to the other osmose bug tag. So if I want to fix one such problem, I have to open the osmose tags total of 4 times, two times to save and two times to upload, or so I have understood. What exactly does the "upload" do? If I close a osmose bug, and when I upload a changeset normally, will it do the same thing as pressing the "upload" button in the osmose problems popup window?

Bonus question: Why some tags don't have some common presets, like for example if I choose/type "religion" as the key, it suggests me the following values only: Bahai, Caodaism, Confucian, Jain, Sikh, Spiritualist, Zoroastrian, Taoist, Tenrikyo and Unitarian universalist. Here none of those apply, and most are christian, but it also seems to miss all the other common options like muslim, jewish, hindu, buddhist.

 By the way is there a chatroom with vespucci users? I have tried asking some vespucci questions on the "main" OSM discord/matrix server at OSM Software room but probably there is a better place?

**Comment by simonpoole** [comment]
Starting at the end, the main OSM channel, that is #osm-talk and the corresponding matrix room is just fine, if the volume gets too high we can always create a special purpose room.

WRT the religion tag, some of the presets have reduced scope religion values, so it really depends on what you where trying to tag, in any case the discussion / issue belongs in https://github.com/simonpoole/beautified-JOSM-preset

Osmose:  you don't "have to" close the issues as they will eventually disappear when the data is fixed (may need a refresh of the data a couple of days later in Vespucci as we cache the data). Naturally not closing the issues has a number of downsides, so improving the workflow could be a thing to do. Down the road there is some refactoring planned in any case as we are still using the old Osmose API.

**Comment by simonpoole** [comment]
Fixed in https://github.com/MarcusWolschon/osmeditor4android/pull/1651
